vault/core.go

```diff
@@ -652,6 +652,16 @@ func (c *Core) fetchACLandTokenEntry(req *logical.Request) (*ACL, *TokenEntry, e
 		return nil, nil, logical.ErrPermissionDenied
 	}
 
+	// If we are still restoring the expiration manager, we want to ensure the
```

Comment:
I think this should happen in the token store's lookupSalted. I don't think the token store should ever return a token entry on any lookup anywhere for a token that is expired.

Reply:
I will move.